### PR TITLE
Task/NLU-2697 - 3,000,000 detected as 3 in "es-mx" MS Recognizers library

### PR DESCRIPTION
**Pull request by Conor-Keaney**

### CODE REVIEW

File: Python/libraries/recognizers-number-with-unit/recognizers_number_with_unit/number_with_unit/number_with_unit_recognizer.py [hunk 1]
```diff
@@ -201,16 +201,20 @@ def initialize_configuration(self):
 
         # region Spanish Mexican
         self.register_model('CurrencyModel', Culture.SpanishMexican, lambda options: CurrencyModel(
-            [ExtractorParserModel(BaseMergedUnitExtractor(SpanishCurrencyExtractorConfiguration(
-            )), BaseMergedUnitParser(SpanishCurrencyParserConfiguration(culture_info=CultureInfo(Culture.SpanishMexican))))]
+            [ExtractorParserModel(
+                BaseMergedUnitExtractor(SpanishCurrencyExtractorConfiguration(CultureInfo(Culture.SpanishMexican)
```
Comment (ushmau5) [inline Python/libraries/recognizers-number-with-unit/recognizers_number_with_unit/number_with_unit/number_with_unit_recognizer.py:205]: I see the `culture_info` is passed as a keyword argument to the Configuration below this. Maybe we should keep the same pattern Microsoft are already following if possible? Looks like it is just missing here.

File: Python/libraries/recognizers-number-with-unit/recognizers_number_with_unit/number_with_unit/number_with_unit_recognizer.py [hunk 2]
```diff
@@ -258,26 +262,30 @@ def get_temperature_model(self, culture: str = None, fallback_to_default_culture
         return self.get_model('TemperatureModel', culture, fallback_to_default_culture)
 
 
-def recognize_age(query: str, culture: str, options: NumberWithUnitOptions = NumberWithUnitOptions.NONE, fallback_to_default_culture: bool = True) -> List[ModelResult]:
+def recognize_age(query: str, culture: str, options: NumberWithUnitOptions = NumberWithUnitOptions.NONE,
```
Comment (ushmau5) [inline Python/libraries/recognizers-number-with-unit/recognizers_number_with_unit/number_with_unit/number_with_unit_recognizer.py:265]: Same here again I don't see any changes here these are just formatting changes right?

File: Python/libraries/recognizers-number/recognizers_number/number/spanish/extractors.py [hunk 3]
```diff
@@ -24,19 +25,19 @@ def ambiguity_filters_dict(self) -> List[ReRe]:
...
             cardinal_ex = SpanishCardinalExtractor(
-                SpanishNumeric.PlaceHolderPureNumber)
+                SpanishNumeric.PlaceHolderPureNumber, culture)
         elif mode is NumberMode.CURRENCY:
             self.__regexes.append(
                 ReVal(re=SpanishNumeric.CurrencyRegex, val='IntegerNum'))
 
         if cardinal_ex is None:
-            cardinal_ex = SpanishCardinalExtractor()
+            cardinal_ex = SpanishCardinalExtractor(culture=culture)
```
Comment (ushmau5) [inline Python/libraries/recognizers-number/recognizers_number/number/spanish/extractors.py:40]: We don't use cardinals so might be able to leave this out.

File: Python/libraries/recognizers-number/recognizers_number/number/spanish/extractors.py [hunk 4]
```diff
@@ -83,7 +84,11 @@ def regexes(self) -> List[
     def _extract_type(self) -> str:
         return Constants.SYS_NUM_INTEGER
 
-    def __init__(self, placeholder: str = SpanishNumeric.PlaceHolderDefault):
+    def __init__(self, placeholder: str = SpanishNumeric.PlaceHolderDefault, culture=Culture.Spanish):
+        thousandMarker = LongFormatMode.INTEGER_COMMA
```
Comment (ushmau5) [inline Python/libraries/recognizers-number/recognizers_number/number/spanish/extractors.py:88]: MS are using underscore naming convention so would be better to go with `thousand_marker`. They also reference this idea as a `thousands_separator` if you want to follow their terminology (https://docs.microsoft.com/en-us/globalization/locale/number-formatting)

File: Python/libraries/recognizers-number-with-unit/recognizers_number_with_unit/number_with_unit/number_with_unit_recognizer.py (hunk 2 above)
Comment (Conor-Keaney) [reply Python/libraries/recognizers-number-with-unit/recognizers_number_with_unit/number_with_unit/number_with_unit_recognizer.py:265]: Yeah thats right, I believe these formatting changes were just auto done when I had accepted one, i'll revert them

File: Specs/Number/SpanishMexican/NumberModel.json [hunk 5]
```diff
@@ -166,5 +166,21 @@
         "End": 17
       }
     ]
+  },
```
[the remark this answers is not in the document]
Comment (srandal) [reply Specs/Number/SpanishMexican/NumberModel.json:169]: And maybe add it here as well. From the other test cases, it looks like expected behavior for SpanishMexican is to resolve as a decimal-comma system if it doesn't fit a legitimate decimal-point pattern and it does fit a decimal-comma pattern (e.g. "0,683" "1.111,11").

**Comment by samhickey25** [comment]
Just so everyone knows, this bugfix was originally just for amountOfMoney, but Conor has also fixed the issue for number, which is an improvement on what LUIS currently offers

File: Python/libraries/recognizers-number/recognizers_number/number/spanish/extractors.py (hunk 3 above)
Comment (Conor-Keaney) [reply Python/libraries/recognizers-number/recognizers_number/number/spanish/extractors.py:40]: Seems to still be in use, if its anything other than PURE_NUMBER then its still called, removing it leads to "3.000.000" to fail

File: Specs/Number/SpanishMexican/NumberModel.json (hunk 5 above)
[the remark this answers is not in the document]
Comment (Conor-Keaney) [reply Specs/Number/SpanishMexican/NumberModel.json:169]: Oh sorry yes good point, what i was talking about was line 145, when there is both a dot and comma does it work - 1.111,11 and 1,111.11 both = 1111.11. 
I see what you mean, i'd say we might leave it for now but I'll make note of it for later
Thanks Scott!

**Review by kevinwalshgen — APPROVED** [review]

**Comment by ushmau5** [comment]
Do we need to bump package versions here? How will these changes be pulled into our service?

**Review by srandal — APPROVED** [review]

**Review by colm-dillon — APPROVED** [review]

**Review by andrew-gradinari — APPROVED** [review]

**Pull request merged**